### ISSUE TITLE
Improve presentation of inline suggestions with inprecise prefix matches

### ISSUE THREAD

**Issue by hediet** [microsoft/vscode#147068]

**Comment by jrieken** [comment]
One suggestion would be to remove the underline. It conflicts with squiggle (which is often present too) and I don't see the added value of it. 

<img width="167" alt="Screenshot 2022-04-13 at 11 40 57" src="https://user-images.githubusercontent.com/1794099/163148211-7161db46-52f2-435e-8099-ed91191b1cc0.png">

**Comment by jrieken** [comment]
Also an idea could be to highlight those parts of the prefix that do match and how they match. The mock-up below shows show `createE` could be matched against `createElement` (the sample is fake, best fuzzy match would be _create_Element

<img width="385" alt="Screenshot 2022-04-13 at 11 57 32" src="https://user-images.githubusercontent.com/1794099/163153692-761240d8-3930-4038-a8dc-4f814ff41954.png">

**Comment by jrieken** [comment]
Last option could be to be more strict, e.g never show inprecise prefix matches, e.g don't show `console` for `coN`, only for `con`

**Comment by misolori** [comment]
What's the value in showing in-precise prefix matches? We don't currently do this in intellisense suggestions:

<img width="179" alt="CleanShot 2022-04-18 at 12 22 09@2x" src="https://user-images.githubusercontent.com/35271042/163863868-a3f71517-d61e-42d0-afa8-28eb0be44063.png">

**Comment by jrieken** [comment]
We fuzzy matching, like `futio` matches `function` and others. For IntelliSense the rule of thumb is that all typed characters must match the suggestion in the same order (there is also an exception from that role but that's the general idea)

<img width="524" alt="Screenshot 2022-04-19 at 08 30 49" src="https://user-images.githubusercontent.com/1794099/163940075-7c1cbf47-dff7-4f66-9dfa-e6cad79a4a28.png">

**Comment by misolori** [comment]
I guess I still struggle with the **value** of showing in-precise matches here. Anything that is not a continuation of what the user is intending to write (i.e. showing the full match inline) seems to slow down the workflow.

**Comment by misolori** [comment]
One alternate idea, instead of showing additional characters inline, what if we bolded the matches of the text that is there already? Might be trickier with fonts & themes that use boldness but would at least prevent introducing additional items for the user to process:

<img width="279" alt="CleanShot 2022-04-19 at 08 54 10@2x" src="https://user-images.githubusercontent.com/35271042/164045317-63dc7e35-a6c0-4801-b2f2-d5672a3ea5ca.png">

**Comment by jrieken** [comment]
Yeah, that's basically this option https://github.com/microsoft/vscode/issues/147068#issuecomment-1097852540, only continue strict, e.g case sensitive, prefixes. I believe that's fair to begin with and how other tools do it too. 

On the question of "value": often you have the same prefix repeated, e.g `createElement` and `createAttribute` or `getA` and `getB` and `getC`. When coding I often know this and what to "jump ahead" by typing `creAtt`. Tho, the question is if that's something for inline completions or if that a case for the suggest widget only?

**Comment by byehack** [comment]
related https://github.com/microsoft/vscode/issues/130297
